shell/platform/darwin/common/framework/Source/FlutterStandardCodec.mm

```diff
@@ -221,115 +230,142 @@ - (void)dealloc {
+    WriteByte(data, 255);
+    WriteBytes(data, &size, 4);
   }
 }
 
-- (void)writeAlignment:(UInt8)alignment {
-  UInt8 mod = _data.length % alignment;
+static void WriteAlignment(CFMutableDataRef data, UInt8 alignment) {
+  assert(alignment <= 8);
+  UInt8 mod = CFDataGetLength(data) % alignment;
   if (mod) {
-    for (int i = 0; i < (alignment - mod); i++) {
-      [self writeByte:0];
-    }
+    WriteBytes(data, s_zeroBuffer, alignment - mod);
   }
 }
 
-- (void)writeUTF8:(NSString*)value {
+static void WriteUTF8(CFMutableDataRef data, NSString* value) {
   UInt32 length = [value lengthOfBytesUsingEncoding:NSUTF8StringEncoding];
-  [self writeSize:length];
-  [self writeBytes:value.UTF8String length:length];
+  WriteSize(data, length);
+  WriteBytes(data, value.UTF8String, length);
 }
 
-- (void)writeValue:(id)value {
-  if (value == nil || value == [NSNull null]) {
-    [self writeByte:FlutterStandardFieldNil];
-  } else if ([value isKindOfClass:[NSNumber class]]) {
+static void WriteValue(CFMutableDataRef data, id value) {
+  if (value == nil || value == s_nsNull) {
+    WriteByte(data, FlutterStandardFieldNil);
+  } else if ([value isKindOfClass:s_nsNumber]) {
     CFNumberRef number = (CFNumberRef)value;
     BOOL success = NO;
     if (CFGetTypeID(number) == CFBooleanGetTypeID()) {
       BOOL b = CFBooleanGetValue((CFBooleanRef)number);
-      [self writeByte:(b ? FlutterStandardFieldTrue : FlutterStandardFieldFalse)];
+      WriteByte(data, (b ? FlutterStandardFieldTrue : FlutterStandardFieldFalse));
       success = YES;
     } else if (CFNumberIsFloatType(number)) {
       Float64 f;
       success = CFNumberGetValue(number, kCFNumberFloat64Type, &f);
       if (success) {
-        [self writeByte:FlutterStandardFieldFloat64];
-        [self writeAlignment:8];
-        [self writeBytes:(UInt8*)&f length:8];
+        WriteByte(data, FlutterStandardFieldFloat64);
+        WriteAlignment(data, 8);
+        WriteBytes(data, (UInt8*)&f, 8);
       }
     } else if (CFNumberGetByteSize(number) <= 4) {
       SInt32 n;
       success = CFNumberGetValue(number, kCFNumberSInt32Type, &n);
       if (success) {
-        [self writeByte:FlutterStandardFieldInt32];
-        [self writeBytes:(UInt8*)&n length:4];
+        WriteByte(data, FlutterStandardFieldInt32);
+        WriteBytes(data, (UInt8*)&n, 4);
       }
     } else if (CFNumberGetByteSize(number) <= 8) {
       SInt64 n;
       success = CFNumberGetValue(number, kCFNumberSInt64Type, &n);
       if (success) {
-        [self writeByte:FlutterStandardFieldInt64];
-        [self writeBytes:(UInt8*)&n length:8];
+        WriteByte(data, FlutterStandardFieldInt64);
+        WriteBytes(data, (UInt8*)&n, 8);
       }
     }
     if (!success) {
       NSLog(@"Unsupported value: %@ of number type %ld", value, CFNumberGetType(number));
-      NSAssert(NO, @"Unsupported value for standard codec");
+      assert(NO);  // Unsupported value for standard codec.
```

Comment:
Can we leave the `NSAssert` since the nicer `NSAssertionHandler` behavior with file and line numbers, etc would be more important than performance at the point it fails?

Reply:
You can't use NSAssert outside of objective-c methods, it's a macro that depends on having access to `self`.

Reply:
Oh right, `NSCAssert` then.

Reply:
ahh yea, forgot about that, thanks, done